hive/src/main/java/com/netease/arctic/hive/utils/HiveMetaSynchronizer.java

```diff
@@ -92,32 +92,39 @@ public static void syncHiveSchemaToArctic(ArcticTable table, HMSClientPool hiveC
   private static boolean updateStructSchema(TableIdentifier tableIdentifier, UpdateSchema updateSchema,
       String parentName, Types.StructType icebergStruct, Types.StructType hiveStruct) {
     boolean update = false;
-    for (int i = 0; i < hiveStruct.fields().size(); i++) {
-      Types.NestedField hiveField = hiveStruct.fields().get(i);
-      Types.NestedField icebergField = icebergStruct.field(hiveField.name());
-      if (icebergField == null) {
+    for (Types.NestedField hiveField : hiveStruct.fields()) {
+      // to check if lowercase matches
+      List<Types.NestedField> fields = icebergStruct
+          .fields()
+          .stream()
+          .filter(f -> f.name().toLowerCase().equals(hiveField.name())).collect(Collectors.toList());
+      if (fields.isEmpty()) {
         updateSchema.addColumn(parentName, hiveField.name(), hiveField.type(), hiveField.doc());
         update = true;
         LOG.info("Table {} sync new hive column {} to arctic", tableIdentifier, hiveField);
-      } else if (!icebergField.type().equals(hiveField.type()) ||
-          !Objects.equals(icebergField.doc(), (hiveField.doc()))) {
-        if (hiveField.type().isPrimitiveType() && icebergField.type().isPrimitiveType()) {
-          if (TypeUtil.isPromotionAllowed(icebergField.type().asPrimitiveType(), hiveField.type().asPrimitiveType())) {
+      } else {
+        Types.NestedField icebergField = fields.get(0);
```

Comment:
If multiple fields are matched, I think an error should be thrown instead of using the first one directly.